source/Calamari.AzureAppService/Behaviors/TargetDiscoveryBehaviour.cs

```diff
@@ -102,101 +85,46 @@ public async Task Execute(RunningDeployment runningDeployment)
-                    },
-                    (ex, delay, retryAttempt, context) =>
-                    {
-                        Log.Verbose($"An error has occurred {description}: {ex.Message}, retrying {retryAttempt} of {maxRetries} after {delay}");
-                    });
-        }
-
-        private IEnumerable<IDeploymentSlot> ListDeploymentSlots(IWebApp webApp)
-        {
-            var retryPolicy = CreateAzureQueryRetryPolicy(5, $"listing deployment slots for web app {webApp.Name}");
-
-            // We could get a 404 listing slots if the web app gets deleted 
-            // after being found but before we can check it's slots, in this
-            // case we'll log a message and continue
-            var webAppNotFoundPolicy = Policy<IEnumerable<IDeploymentSlot>>
-                .Handle<DefaultErrorResponseException>(dex => dex.Response.StatusCode == System.Net.HttpStatusCode.NotFound)
-                .Fallback(
-                    fallbackValue: Enumerable.Empty<IDeploymentSlot>(),
-                    onFallback: (exception, context) => Log.Verbose($"Could not list deployment slots for web app {webApp.Name} as it could no longer be found")
-                );
-            
-            return retryPolicy.Wrap(webAppNotFoundPolicy).Execute(() =>
-            {
-                return webApp.DeploymentSlots.List();
-            });
-        }
-
         private void WriteTargetCreationServiceMessage(
-            IWebApp webApp,
+            AzureResource resource,
             TargetDiscoveryContext<AccountAuthenticationDetails<ServicePrincipalAccount>> context,
             TargetMatchResult matchResult)
         {
-            Log.WriteServiceMessage(
-                TargetDiscoveryHelpers.CreateWebAppTargetCreationServiceMessage(
-                    webApp.ResourceGroupName,
-                    webApp.Name,
-                    context.Authentication!.AccountId,
-                    matchResult.Role,
-                    context.Scope!.WorkerPoolId));
-        }
+            var resourceName = resource.Name;
+            string? slotName = null;
+            // As of version 2022-03-01 of the Resource Details endpoint, the slotName property returned from
+            // Azure is always null, meaning we have to calculate it manually. The slot also does not contain
+            // a property for the WebAppName. Fortunately the name of a slot resource is {webAppName}/{slotName}
+            // so the two required elements can be extracted.
+            if (resource.Type.Equals(AzureRestClient.WebAppSlotsType, StringComparison.InvariantCultureIgnoreCase))
+            {
+                var indexOfSlash = resourceName.LastIndexOf("/", StringComparison.InvariantCulture);
+                if (indexOfSlash >= 0)
+                {
+                    slotName = resourceName[(indexOfSlash + 1)..];
+                    resourceName = resourceName[..indexOfSlash];
+                }
+            }
```

Comment:
This was the best way I could get the Web App Name and the Slot Name from a Slot Resource. I'm not sure if this is too brittle though. I don't think there is really a better option though unfortunately.

Reply:
Given we need to make an additional api call to get the resource group on each discovered resource anyway, I wonder if its worth seeing if we could use the client to query the webapp/slot resource properties in a more typed way, that way we could have more safety around getting the slot name as well as the other resource properties?

Reply:
I just went on another voyage of discovery with this library. I can actually get all the info I need from the `Id` property - of type `ResourceIdentifier`. So we don't actually need a second call at all, and I don't need to parse the name into web app name and slot name! 👍